### ISSUE TITLE
Many methods return lifetimes that are not tied to the lifetime of self

### ISSUE THREAD

**Issue by goffrie** [al8n/caches-rs#11]
These use `&'_ self` when they should use `&'a self` (resp. `mut`).

Sample program that segfaults when it shouldn't compile:
```rust
//! ```cargo
//! [dependencies]
//! caches = { version = "0.2" }
//! ```

use caches::{Cache, RawLRU};

fn main() {
    let mut lru = RawLRU::new(2).unwrap();
    lru.put(1, "foo".to_string());
    lru.put(2, "bar".to_string());
    let iter = lru.iter();
    drop(lru);
    println!("{:?}", iter.collect::<Vec<_>>());
}
```

**Comment by goffrie** [comment]
Actually it's not just RawLRU - all the methods on `trait Cache` that return references have the same problem.

**Comment by al8n** [comment]
Hi, thanks for reporting. Fix the lifetime bug in `0.2.6` error.
